### PR TITLE
[DataFrame] Implements DataFrame.rename, DataFrame.rename_axis, and Index.set_names

### PR DESCRIPTION
**Pull request by SaladRaider**
## What do these changes do?

Implements DataFrame.rename, DataFrame.rename_axis, and Index.set_names with tests.

**NOTE:** 
* This is branched off of PR #1545 in order to get the ray dataframe index functionality
* Tests on transpose fails on the version I branched off from

### CODE REVIEW

File: .gitignore
Comment (devin-petersohn) [inline .gitignore]: @robertnishihara Thoughts on these additions? I don't think they're necessary, but I'll let you make the call

**Comment by devin-petersohn** [comment]
@SaladRaider Please rebase on current master and push to reflect the new changes.

File: .gitignore
Comment (robertnishihara) [reply .gitignore]: Seems fine. If it's an issue we can always revert the changes to the `.gitignore`.

File: python/ray/dataframe/dataframe.py
Comment (robertnishihara) [inline python/ray/dataframe/dataframe.py]: Minor, but why separate `import warnings` from the other imports?

**Comment by devin-petersohn** [comment]
@SaladRaider if you wait until #1598 is merged, these implementations should become easier.

File: python/ray/dataframe/dataframe.py
Comment (devin-petersohn) [inline python/ray/dataframe/dataframe.py]: Why does this need to be a remote task? I think you can just call `list(df.index)` and it will be faster.

File: python/ray/dataframe/dataframe.py
Comment (devin-petersohn) [inline python/ray/dataframe/dataframe.py]: Please remove dead code. Also, in the parameters remove the `sorted` parameter.

File: python/ray/dataframe/index.py
Comment (devin-petersohn) [inline python/ray/dataframe/index.py]: This file was removed in a previous commit.

**Review by devin-petersohn — CHANGES_REQUESTED** [review]
Just a few minor changes. Thanks!

**Comment by devin-petersohn** [comment]
@robertnishihara Please take a quick review pass on this PR since I have touched the code.

**Review by devin-petersohn — APPROVED** [review]

File: python/ray/dataframe/dataframe.py
Comment (robertnishihara) [inline python/ray/dataframe/dataframe.py]: Move this sentence to previous line like

```
"""Alter the name or names of the axis.
```

**Review by robertnishihara — APPROVED** [review]

**Comment by devin-petersohn** [comment]
Test failures unrelated, ready to merge.

**Pull request merged**
